### PR TITLE
Update YamlCreate.ps1 (Listen For Keypress)

### PR DESCRIPTION
**Pull request by vedantmgoyal9**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----
From #21213

@OfficialEsco Sorry for that mistake. Here is a working fix.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/21258)

### CODE REVIEW

**Comment by OfficialEsco** [comment]
Personally i don't see the reason to use `Yaml.Create` in those PRs, its not matching wingetcreate and none else is using it but you

**Comment by ImJoakim** [comment]
> Personally i don't see the reason to use `Yaml.Create` in those PRs, its not matching wingetcreate and none else is using it but you

I agree with this. In any case, you can make that change on your local machine and use it yourself.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
https://github.com/microsoft/winget-pkgs/pull/21258#issuecomment-881859853

**Comment by vedantmgoyal2009** [comment]
Made some more changes from my side.

**Review by OfficialEsco — CHANGES_REQUESTED** [review]
```
[Optional] Enter the application product code. Looks like {CF8E6E00-9C03-4440-81C0-21FACB921A6B}
ProductCode found from installer:
Can be found with get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize
ProductCode:
Scope
[Optional] Enter the Installer Scope.
[M] Machine  [U] User  [N] No idea (default is 'N'): uUpgradeBehavior
[Optional] Enter the UpgradeBehavior.
[I] install  [U] uninstallPrevious (default is 'I'): i
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No  [?] Help (default is "N"):
```

* Needs more spacing
* UpgradeBehaviour ends up on the same line as the Scope input field
* UpgradeBehaviour got ignored
image

**Comment by vedantmgoyal2009** [comment]
Needs more spacing - Fixed
UpgradeBehaviour ends up on the same line as the Scope input field - Fixed
UpgradeBehaviour got ignored  - Fixed

Working on few more changes, will commit the new changes soon. 🙂

**Comment by OfficialEsco** [comment]
Not sure how you managed to do this but when i entered the InstallModes it looped me
```
[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:

[Optional] Enter any Protocols the application provides a handler for. For example: http, https (Max 16)
Protocols:

[Optional] Enter any Commands or aliases to run the application. For example: msedge (Max 16)
Commands:

[Optional] List of additional non-zero installer success exit codes other than known default values by winget (Max 16)
InstallerSuccessCodes:

[Optional] List of supported installer modes. Options: interactive, silent, silentWithProgress
InstallModes: interactive, silent

[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:

[Optional] Enter any Protocols the application provides a handler for. For example: http, https (Max 16)
Protocols:

[Optional] Enter any Commands or aliases to run the application. For example: msedge (Max 16)
Commands:

[Optional] List of additional non-zero installer success exit codes other than known default values by winget (Max 16)
InstallerSuccessCodes:

[Optional] List of supported installer modes. Options: interactive, silent, silentWithProgress
Old Variable: interactive, silent
InstallModes:

[Optional] Enter the full publisher name. For example: Microsoft Corporation
Old Variable: Wiris
Publisher:
```

**Comment by vedantmgoyal2009** [comment]
Thanks for your feedback. I will into it and push a fix soon.

**Comment by vedantmgoyal2009** [comment]
I have fixed the looping issue. If there is any issue now, please tell me. I have tried my best to reduce the number of times a user has to press enter↵ while creating a manifest. I have implemented Trenly's logic from #21143.

**Comment by OfficialEsco** [comment]
@quhxl can you test it?

**Comment by quhxl** [comment]
Before:
before
After:
after

@OfficialEsco Seems to be fixed. @vedantmgoyal2009 But I also got looped at `FileExtensions` to `InstallMode`.

**Comment by vedantmgoyal2009** [comment]
Thanks, I will look into the looping issue again.

File: Tools/YamlCreate.ps1
Comment (OfficialEsco) [inline Tools/YamlCreate.ps1:418]:
```suggestion
        Write-Host; Read-WinGet-InstallerValues
```
```
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No (default is 'N'): y


[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:
```
This is inverted, If Yes loop back

**Comment by vedantmgoyal2009** [comment]
I have fixed it using another way. Now, it will work in any case.

**Review by OfficialEsco — CHANGES_REQUESTED** [review]

File: Tools/YamlCreate.ps1
Comment (vedantmgoyal2009) [reply Tools/YamlCreate.ps1:418]: You mean the issue is fixed now?

Comment (OfficialEsco) [reply Tools/YamlCreate.ps1:418]: This suggestion is still Inverted

My other issue seems to be solved, not sure where quhxl got looped since he didn't provide log

Comment (OfficialEsco) [reply Tools/YamlCreate.ps1]:
```
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No (default is 'N'): n

[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:
```
Nvm

Comment (OfficialEsco) [reply Tools/YamlCreate.ps1]: the last 2 commits did indeed solve the issue

Comment (vedantmgoyal2009) [reply Tools/YamlCreate.ps1]: I have committed changes.

**Comment by OfficialEsco** [comment]
Seems to be stable for me now

This part needs a bit more space
```
Save to disk?
Do you want to save the files to the Temp folder?
[Y] Yes  [N] No  [M] Manually Enter SHA256 (default is 'N'):
Downloading URL. This will take a while...
Time taken: 3 second(s)
```

This part got one space too many 😁
```
[Optional] Enter the application product code. Looks like {CF8E6E00-9C03-4440-81C0-21FACB921A6B}
ProductCode found from installer:
Can be found with get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize
ProductCode:


Scope
[Optional] Enter the Installer Scope.
[M] Machine  [U] User  [N] No idea (default is 'N'): u
```

And this part needs a bit more space
```
Yaml file created: C:\Users\odin9\GitHub\winget-pkgs\manifests\e\Eugeny\Tabby\1.0.147\Eugeny.Tabby.locale.en-US.yaml
Manifest validation succeeded.
Submit PR?
Do you want to submit your PR now?
Switched to a new branch 'Eugeny.Tabby-1.0.147'
```

**Comment by vedantmgoyal2009** [comment]
I think now, it is working perfectly.

**Comment by vedantmgoyal2009** [comment]
Sorry, by mistake, committed manifests to this branch.

**Comment by denelon** [comment]
@OfficialEsco I'll wait to review/approve/merge until after you validate and approve.

**Review by OfficialEsco — APPROVED** [review]
Seems to be working correctly in 3 Updates and 1 New Manifest, i have not tested the Sandbox prompt tho

**Review by denelon — APPROVED** [review]

**Pull request merged**
